lib/puppet/type/archive.rb

```diff
@@ -103,7 +103,7 @@ def should_to_s(value)
   newparam(:source) do
     desc 'archive file source, supports http|https|ftp|file uri.'
     validate do |value|
-      unless value =~ URI.regexp(%w(http https file ftp))
+      unless value =~ URI.regexp(%w(http https file ftp)) || value =~ %r{^([A-Z]:|\/)}
```

Comment:
It looks like the function has support for Windows already. Do you mean something else?

https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util.rb#L237